File: src/week1/P007.py
# 1874
n = int(input())
stack = []
result = []
flag = 0
cur = 1
for i in range(n):
    num = int(input())
    while cur <= num:
        stack.append(cur)
        result.append("+")
        cur += 1

    if stack[-1] == num:
        stack.pop()
        result.append("-")
    else:
        print("NO")
        flag = 1
        break

if flag == 0:
    for i in result:
        print(i)

File: src/week1/P007.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n').map(Number);

const n = input[0];
const sequence = input.slice(1);

const stack = [];
const result = [];
let currentNum = 1;
let isPossible = true;

for (let i = 0; i < n; i++) {
    const targetNum = sequence[i];

    while (currentNum <= targetNum) {
        stack.push(currentNum);
        result.push('+');
        currentNum++;
    }

    if (stack[stack.length - 1] === targetNum) {
        stack.pop();
        result.push('-');
    } else {
        isPossible = false;
        break;
    }
}

if (isPossible) {
    console.log(result.join('\n'));
} else {
    console.log('NO');
}